curp-external-api/src/cmd.rs

```diff
@@ -11,6 +11,16 @@ use crate::LogIndex;
 pub trait Command:
     Sized + Sync + Send + DeserializeOwned + Serialize + std::fmt::Debug + Clone + ConflictCheck
 {
+    /// The error type that is returned in execution
```

Comment:
I guess the main reason might be to obtain user-defined error types in some related methods of `ConnectApi` trait. It may look weird to introduce a dependency between `CommandExecutor` and `ConnectApi`. `Command` seems more appropriate. So, the Error type was moved from CE to Command. After all, there's no need to keep the error type in both Command and CE.